lib/services/client_data_creator.rb

```diff
@@ -0,0 +1,41 @@
+class ClientDataCreator
+  def initialize(client_data, user, attachment_data = [])
+    @client_data = client_data
+    @user = user
+    @attachment_data = attachment_data
+  end
+
+  def run
+    client_data.build_proposal(flow: 'linear', requester: user)
```

Comment:
this is also being done in the controller because there is a validation on client_data for existence of a proposal. But I also like doing this here because otherwise in my test setup I need to build the proposal for the client data object, which seems like a smell. What do you think?